src/Microsoft.Extensions.Configuration.FileProviderExtensions/FileConfigurationProvider.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public abstract class FileConfigurationProvider : ConfigurationProvider
```

Comment:
Assuming taking a dependency on FileProviders is ok we should consider moving this class together with the stuff in Microsoft.Extensions.Configuration.FileExtensions into the same library, then rename it to Microsoft.Extensions.Configuration.FileBased, FileProviders, or something similar.